Makefile

```diff
@@ -38,82 +38,100 @@ DIFF = $(shell git diff --quiet >/dev/null 2>&1; if [ $$? -eq 1 ]; then echo "1"
+LDFLAGS=-X $(PKG).GitVersion=$(GIT_VERSION) -X $(PKG).gitCommit=$(GIT_HASH) -X $(PKG).gitTreeState=$(GIT_TREESTATE) -X $(PKG).buildDate=$(BUILD_DATE)
 
-LDFLAGS="-X $(PKG).GitVersion=$(GIT_VERSION) -X $(PKG).gitCommit=$(GIT_HASH) -X $(PKG).gitTreeState=$(GIT_TREESTATE) -X $(PKG).buildDate=$(BUILD_DATE)"
+SRCS = $(shell find cmd -iname "*.go") $(shell find pkg -iname "*.go")
 
-.PHONY: all lint test clean cosign cross
+GOLANGCI_LINT_DIR = $(shell pwd)/bin
+GOLANGCI_LINT_BIN = $(GOLANGCI_LINT_DIR)/golangci-lint
 
+.PHONY: all lint test clean cosign cross
 all: cosign
 
-help: # Display help
-	@awk -F ':|##' \
-		'/^[^\t].+?:.*?##/ {\
-			printf "\033[36m%-30s\033[0m %s\n", $$1, $$NF \
-		}' $(MAKEFILE_LIST) | sort
-
-SRCS = $(shell find cmd -iname "*.go") $(shell find pkg -iname "*.go")
-
 cosign: $(SRCS)
-	CGO_ENABLED=0 go build -trimpath -ldflags $(LDFLAGS) -o $@ ./cmd/cosign
+	CGO_ENABLED=0 go build -trimpath -ldflags "$(LDFLAGS)" -o $@ ./cmd/cosign
 
 cosign-pivkey: $(SRCS)
 	CGO_ENABLED=1 go build -trimpath -tags=pivkey -ldflags $(LDFLAGS) -o cosign ./cmd/cosign
 
 cosign-pkcs11key: $(SRCS)
 	CGO_ENABLED=1 go build -trimpath -tags=pkcs11key -ldflags $(LDFLAGS) -o cosign ./cmd/cosign
 
-GOLANGCI_LINT_DIR = $(shell pwd)/bin
-GOLANGCI_LINT_BIN = $(GOLANGCI_LINT_DIR)/golangci-lint
-golangci-lint:
-	rm -f $(GOLANGCI_LINT_BIN) || :
-	set -e ;\
-	GOBIN=$(GOLANGCI_LINT_DIR) go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.42.0 ;\
-
-lint: golangci-lint ## Run golangci-lint linter
-	$(GOLANGCI_LINT_BIN) run -n
 
+.PHONY: cosigned
+cosigned: ## Build cosigned binary
+	CGO_ENABLED=0 go build -trimpath -ldflags $(LDFLAGS) -o $@ ./cmd/cosign/webhook
 
-PLATFORMS=darwin linux windows
-ARCHITECTURES=amd64
+.PHONY: sget
+sget: ## Build sget binary
+	go build -trimpath -o $@ ./cmd/sget
 
+.PHONY: cross
 cross:
 	$(foreach GOOS, $(PLATFORMS),\
 		$(foreach GOARCH, $(ARCHITECTURES), $(shell export GOOS=$(GOOS); export GOARCH=$(GOARCH); \
 	go build -trimpath -ldflags $(LDFLAGS) -o cosign-$(GOOS)-$(GOARCH) ./cmd/cosign; \
 	shasum -a 256 cosign-$(GOOS)-$(GOARCH) > cosign-$(GOOS)-$(GOARCH).sha256 ))) \
 
+
+#####################
+# lint / test section
+#####################
+
+golangci-lint:
+	rm -f $(GOLANGCI_LINT_BIN) || :
+	set -e ;\
+	GOBIN=$(GOLANGCI_LINT_DIR) go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.43.0 ;\
+
+lint: golangci-lint ## Run golangci-lint linter
+	$(GOLANGCI_LINT_BIN) run -n
+
 test:
 	go test ./...
 
 clean:
 	rm -rf cosign
+	rm -rf cosigned
+	rm -rf sget
+	rm -rf dist/
 
+##########
+# ko build
+##########
 .PHONY: ko
 ko:
-	# We can't pass more than one LDFLAG via GOFLAGS, you can't have spaces in there.
-	KO_DOCKER_REPO=${KO_PREFIX}/cosign CGO_ENABLED=0 GOFLAGS="-ldflags=-X=$(PKG).gitCommit=$(GIT_HASH)" ko publish --bare \
+	LDFLAGS="$(LDFLAGS)" GIT_HASH=$(GIT_HASH) GIT_VERSION=$(GIT_VERSION) \
+	KO_DOCKER_REPO=${KO_PREFIX}/cosign ko publish --bare \
```

Comment:
Is there a reason you don't want `KO_DOCKER_REPO=${KO_PREFIX} ko publish --base` in these? This will have the same effect as you have now without having to set `KO_DOCKER_REPO` separately each time.

Reply:
`--base` does not exist i just found `--base-import-paths` but the issue is the cosigned image name we build the build path is `cmd/cosign/webhook` so the image will be called `webhook` instead of `cosigned` but then I guess we can use the current approach for this

Reply:
did some changes based on your feedback @imjasonh , can you please take another look?

thanks so much for your time to review this pull request